packages/orchestration/src/utils/progress.js

```diff
@@ -0,0 +1,178 @@
+
+/**
+ * @template T
+ * @typedef {object} ReporterPublication
+ * @property {T} value
+ * @property {EVow<ReporterPublication<T>>} tail
+ */
+
+/**
+ * @template T
+ * @template [A=unknown]
+ * @callback PublicationReducer Called on the value of each linked T. This
+ *   function should perform any side-effects it can when it receives a new
+ *   value, since there is no guarantee that the ProgressRecord tail will be
+ *   prompt (it may take a long time with cross-chain communication) or ever
+ *   fulfill (it may result in a rejection).
+ * @param {T | undefined | null} value `value == null` is a sentinel for the
+ *   final reducer call.
+ * @param {A} accum previous reduction value
+ * @returns {EVow<A | null>} Returning `null` exits the reduction
+ */
+
+/**
+ * Utilities for working with ResultMeta within an async flow.
+ */
+export const progressTrackerAsyncFlowUtils = {
+  /**
+   * Converts a specimen to a Promise. Since the use of Vows is only needed
+   * outside of async-flows, just reject if we encounter one.
+   *
+   * @template T
+   * @param {EVow<T>} specimen
+   * @returns {Promise<Fulfilled<T>>}
+   */
+  promiseFromVow: async specimen => {
+    const maybeVow = await specimen;
+    !isVow(maybeVow) ||
+      assert.Fail`Unexpected Vow ${maybeVow}; only use ResultMeta utilities within async flows (withOrchestration)`;
+
+    return /** @type {Fulfilled<T>} */ (maybeVow);
+  },
+
+  /**
+   * Extracts values from a chained ReporterPublication, calling `reducer` on
+   * each.
+   *
+   * @template T,[A=unknown]
+   * @param {EVow<ReporterPublication<T>>} progressP
+   * @param {PublicationReducer<T, A>} reducer
+   * @param {A} initialAccum accumulator to pass to the reducer on the first
+   *   call
+   * @returns {Promise<A>} the accumulated result
+   */
+  reduceProgressReports: async (progressP, reducer, initialAccum) => {
+    const { promiseFromVow } = progressTrackerAsyncFlowUtils;
+
+    /** @type {ReporterPublication<T> | null} */
+    let nextReport = await promiseFromVow(progressP);
+    let accum = harden(initialAccum);
+
+    while (true) {
+      const reduction = await promiseFromVow(
+        reducer(nextReport?.value ?? null, accum),
+      );
+      harden(reduction);
+
+      if (reduction == null) {
+        nextReport = null;
+      } else {
+        accum = /** @type {A} */ (reduction);
+      }
+      if (nextReport == null) {
+        return accum;
+      }
+
+      nextReport = (await promiseFromVow(nextReport?.tail)) ?? null;
+    }
+  },
+};
+
+harden(progressTrackerAsyncFlowUtils);
+
+/**
+ * A helper for ProgressPublication operations via VowTools.
+ *
+ * @param {Zone} zone
+ * @param {object} powers
+ * @param {Pick<VowTools, 'makeVowKit'>} powers.vowTools
+ */
+export const prepareProgressTracker = (zone, { vowTools: { makeVowKit } }) => {
+  const makeProgressTrackerKit = zone.exoClassKit(
+    'ProgressTrackerKit',
+    undefined,
+    /**
+     * @param {ProgressReport} [initialReport]
```

Comment:
I don't see anything that uses this initialReport param. Can you motivate it? (If I agree it's worth keeping I'll want the motivation documented on this constructor)